src/components/PlanningTable/RowActions.tsx

```diff
@@ -0,0 +1,51 @@
+'use client'
+
+import { MoreHorizontal as DotsHorizontalIcon } from '@ttab/elephant-ui/icons'
+import { type Row } from '@tanstack/react-table'
+
+import {
+  Button,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger
+} from '@ttab/elephant-ui'
+import { Link } from '..'
+import { type Planning } from './data/schema'
+
+
+interface DataTableRowActionsProps {
+  row: Row<Planning>
+}
+
+export function DataTableRowActions({
+  row
+}: DataTableRowActionsProps): JSX.Element {
+  return (
+    <DropdownMenu>
+      <DropdownMenuTrigger asChild>
+        <Button
+          variant="ghost"
+          className="flex h-8 w-8 p-0 data-[state=open]:bg-muted"
+        >
+          <DotsHorizontalIcon className="h-4 w-4" />
+          <span className="sr-only">Open menu</span>
+        </Button>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent align="end" className="w-[160px]">
+        {row.original._source?.['document.meta.core_assignment.rel.deliverable.uuid']
```

Comment:
An empty array of deliverable uuids would enter the loop and output nothing where I would probably expect 'No deliverables' (if that was the intention). In addition, a ternary with long property checks and inline loops inside of rendering makes it difficult to read and understand. Could (or should...) be rewritten as something like:

```
const deliverableUuids = row.original._source?.['document.meta.core_assignment.rel.deliverable.uuid'] || []

return (
  ...
  <DropdownMenuContent align="end" className="w-[160px]">
    {!deliverableUuids.length && 'No deliverables}
    {deliverableUuids.length && deliverableUuids.map(uuid => {
       return <DropdownMenuItem key={uuid>
         <Link ...>
           {uuid}
         </Link>
       </DropdownMenuItem>
    }
  </DropdownMenuContent>
   ...
)
```

Reply:
Good catch!